verilog/preprocessor/verilog_preprocess.h

```diff
@@ -219,6 +235,9 @@ class VerilogPreprocess {
     bool current_branch_condition_met_;
   };
 
+  // Paths to search into during handling `include
+  std::vector<std::string> search_paths_;
```

Comment:
This should be the `FileList` containing defines, incdirs and SystemVerilog files.

A question: The SystemVerilog files will not be needed here, is there a way to make use of it? or remove it?

Reply:
We can change everything as needed.
Maybe we have a sub-type in `FileList` that contain preprocessing information.

```c++
struct FileList {
   struct PreprocessingInfo {
         std::vector<std::string> include_dirs;
         // ... macro defintions
   };
   std::vector<std::string> file_paths;
   PreprocessingInfo preprocessing;
};
```

Then use that type with only the relevant parts:
```c++
   FileList::PreprocessingInfo
```

Might need to change all the places the current `FileList` is used, but it is worthwhile to improve clarity.

Reply:
There is a cycle in dependency problem whenever I try to include `verilog_project.h` in `verilog_preprocess.h`, because the former includes `verilog_analyzer.h`, which also includes `verilog_preprocess.h`. 

Would it be a good idea if i isolates the definition of the `FileList` in a separte file?

Reply:
Sounds good. Put `FileList` in a separate file.